bouncer/shared/fill_or_kill.ts

```diff
@@ -0,0 +1,139 @@
+import { InternalAsset as Asset, InternalAssets as Assets } from '@chainflip/cli';
+import assert from 'assert';
+import {
+  amountToFineAmount,
+  amountToFineAmountBigInt,
+  assetDecimals,
+  decodeDotAddressForContract,
+  isWithinOnePercent,
+  newAddress,
+  observeBalanceIncrease,
+  shortChainFromAsset,
+} from './utils';
+import { requestNewSwap } from './perform_swap';
+import { send } from './send';
+import { getBalance } from './get_balance';
+import { getChainflipApi, observeEvent } from './utils/substrate';
+import { RefundParameters } from './new_swap';
+
+/// Do a swap with unrealistic minimum price so it gets retried and then refunded.
+async function testMinPriceRefund(asset: Asset, amount: number) {
+  const swapAsset = asset === Assets.Usdc ? Assets.Flip : Assets.Usdc;
+  const refundAddress = await newAddress(asset, 'FOK_REFUND');
+  const destAddress = await newAddress(swapAsset, 'FOK');
+  console.log(`Swap destination address: ${destAddress}`);
+
+  const refundBalanceBefore = await getBalance(asset, refundAddress);
+
+  const invalidMinPrice = amountToFineAmount(
+    '99999999999999999999999999999999999999999999999999999',
+    assetDecimals(asset),
+  );
+
+  await using chainflip = await getChainflipApi();
+  const currentRetryDuration = Number(await chainflip.query.swapping.swapRetryDelay());
+  console.log(`Current swap retry delay: ${currentRetryDuration} blocks`);
+
+  const refundParameters: RefundParameters = {
+    retryDurationBlocks: 1.6 * currentRetryDuration, // More than the swap retry delay, so it will be retried once.
+    refundAddress:
+      asset === Assets.Dot ? decodeDotAddressForContract(refundAddress) : refundAddress,
+    minPrice: invalidMinPrice,
+  };
+
+  console.log(`Requesting swap from ${asset} to ${swapAsset} with unrealistic min price`);
+  const swapRequest = await requestNewSwap(
+    asset,
+    swapAsset,
+    destAddress,
+    'FoK_Test',
+    undefined, // messageMetadata
+    0, // brokerCommissionBps
+    false, // log
+    0, // boostFeeBps
+    refundParameters,
+  );
+  const depositAddress = swapRequest.depositAddress;
+  const depositChannelId = swapRequest.channelId;
+
+  const observeDestinationAddress =
+    asset === Assets.Dot ? decodeDotAddressForContract(destAddress) : destAddress;
+
+  const observeSwapScheduled = observeEvent('swapping:SwapScheduled', {
+    test: (event) =>
+      typeof event.data.origin === 'object' &&
+      'DepositChannel' in event.data.origin &&
+      event.data.origin.DepositChannel.channelId === depositChannelId.toString() &&
+      event.data.destinationAddress[shortChainFromAsset(swapAsset)]?.toLowerCase() ===
+        observeDestinationAddress.toLowerCase(),
+  }).event;
+
```

Comment:
These aren't necessarily unique right? 

I can have a bitcoin deposit channel address sending to ETH and a DOT address sending to ETH at the same address, and this code would match on both if the channel id is the same (which is possible since channel id is unique to chain and not global).

why not use the code that the evm deposits uses for SwapScheduled and factor this out? - though, see next comment first

Reply:
Actually, we should use `observeSwapRequested` (from utils) now. The fields you are looking for have been moved to the new `SwapRequested` event.